packages/workspace/src/generators/ci-workflow/__snapshots__/ci-workflow.spec.ts.snap

```diff
@@ -232,4 +232,52 @@ jobs:
 "
 `;
 
+exports[`lib should generate gitlab config 1`] = `
+"image: node:18
+variables:
+  CI: 'true'
+
+# Creating template for DTE agents
+.dte-agent:
+  interruptible: true
+  script:
+    
+    - npm ci
+    - npx nx-cloud start-agent
+
+# Creating template for a job running DTE (orchestrator)
+.base-pipeline:
+  interruptible: true
+  only:
+    - main
+    - merge_requests
+  before_script:
+    
+    - npm ci
+    - NX_HEAD=$CI_COMMIT_SHA
+    - NX_BASE=\${CI_MERGE_REQUEST_DIFF_BASE_SHA:-$CI_COMMIT_BEFORE_SHA}
+
+# Main job running DTE
+CI:
+  stage: affected
+  extends: .base-pipeline
+  script:
+    - npx nx-cloud start-ci-run --stop-agents-after=\\"build\\"
+    - npx nx-cloud record -- yarn nx workspace-lint --base=$NX_BASE --head=$NX_HEAD
+    - npx nx-cloud record -- yarn nx format:check --base=$NX_BASE --head=$NX_HEAD
+    - npx nx affected --base=$NX_BASE --head=$NX_HEAD --target=lint --parallel=3 & yarn nx affected --base=$NX_BASE --head=$NX_HEAD --target=test --parallel=3 --ci --code-coverage & yarn nx affected --base=$NX_BASE --head=$NX_HEAD --target=e2e --parallel=3 --ci --code-coverage & yarn nx affected --base=$NX_BASE --head=$NX_HEAD --target=build --parallel=3
```

Comment:
On Azure I know there are some issues with using `&` and it not waiting for the longest running task to complete, or missing terminal outputs from some of it. Have you verified that this is not the case with gitlab?

Reply:
Yep, this is just adding the Gitlab config that we already have in the docs to the generators. Didn't notice anything like that testing with a clone of the nx-examples repo.